PlayList.java

```diff
@@ -35,32 +35,59 @@ public Track getTrack(int index) {
      *  If the list is full, does nothing and returns false.
      *  Otherwise, appends the track and returns true. */
     public boolean add(Track track) {
-        //// replace the following statement with your code
+        int size1 = this.getSize();
+        if(size1 == this.getMaxSize()){
+            return false;
+        }
+        else{
+            tracks[size1] = track;
+            size = size1 + 1;
+        }
         return true;
     }
 
     /** Returns the data of this list, as a string. Each track appears in a separate line. */
     //// For an efficient implementation, use StringBuilder.
     public String toString() {
-        //// replace the following statement with your code
-        return "";
+        StringBuilder trackString = new StringBuilder();
+        for (int i = 0 ; i < this.getSize(); i++){
+            trackString.append("\n");
+            trackString.append(tracks[i].toString());
+        }
+        return trackString.toString();
     }
 
     /** Removes the last track from this list. If the list is empty, does nothing. */
      public void removeLast() {
-        //// replace this comment with your code
+        if(this.getSize() > 0 ){
+            tracks[size - 1] = null;
+            size = this.getSize() - 1;
+        }
     }
     
     /** Returns the total duration (in seconds) of all the tracks in this list.*/
     public int totalDuration() {
-        //// replace the following statement with your code
-        return 0;
+        int sum = 0 ;
+        for(int i = 0 ; i < this.getSize() ; i++){
+            sum = sum + tracks[i].getDuration();
+        }
+        return sum;
     }
 
     /** Returns the index of the track with the given title in this list.
      *  If such a track is not found, returns -1. */
     public int indexOf(String title) {
-        //// replace the following statement with your code
+        String subtitle = title.substring(1);
+        subtitle = subtitle.toLowerCase();
+        String first = title.substring(0, 1);
+        first = first.toUpperCase();
+        title = first + subtitle;
+
+        for(int i = 0 ; i < this.getSize() ; i++){
```

Comment:
It's better to not call function during loop breaking condition - to avoid that we use variables, however you already got the required property in `this`
```suggestion
        for(int i = 0 ; i < this.size; i++){
```